ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,193 @@
+"""
+High level processing of showers.
+
+This processor will be able to process a shower/event in 3 steps:
+- shower geometry
+- estimation of energy (optional, currently unavailable)
+- estimation of classification (optional, currently unavailable)
+
+"""
+import numpy as np
+
+from astropy.coordinates import SkyCoord, AltAz
+
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import List
+from ctapipe.containers import ArrayEventContainer, ReconstructedShowerContainer
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.reco import HillasReconstructor
+
+
+DEFAULT_SHOWER_PARAMETERS = ReconstructedShowerContainer()
+
+
+class ShowerQualityQuery(QualityQuery):
+    """Configuring shower-wise data checks."""
+
+    quality_criteria = List(
+        default_value=[
+            ("Positive charge", "lambda hillas: hillas.intensity > 0"),
+            ("Positive width", "lambda hillas: hillas.width.value > 0"),
+            ("Positive length", "lambda hillas: hillas.length.value > 0"),
+            ("Ellipticity > 0.1", "lambda hillas: hillas.width.value / hillas.length.value > 0.1"),
+            ("Ellipticity < 0.6", "lambda hillas: hillas.width.value / hillas.length.value < 0.6")
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class ShowerProcessor(Component):
+    """
+    Takes DL1/parameters data and estimates the shower geometry.
+    Should be run after ImageProcessor to produce all DL1b information.
+
+    For now the only supported reconstructor is HillasReconstructor.
+    """
+
+    def __init__(
+        self,
+        subarray: SubarrayDescription,
+        is_simulation,
+        reconstruct_energy,
+        classify,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        subarray: SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        is_simulation: bool
+            If true, also process simulated images if they exist
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
+
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+        self.check_shower = ShowerQualityQuery(parent=self)
+        self._is_simulation = is_simulation
+        self.reconstruct_energy = reconstruct_energy
```

Comment:
This must be traitlets, not kwargs to `__init__`. Otherwise they cannot be set via the configuration.

Reply:
Ops. I didn't test the configuration form the tool, my bad.

Reply:
actually, there is no tool yet...how can I test if this works? I am not really yet used to these traitlets...

Reply:
anyway I gues you mean something like some of the options of the image extractors?

Reply:
like e.g.

```python
apply_integration_correction = BoolTelescopeParameter(
        default_value=True, help="Apply the integration window correction"
    ).tag(config=True)
```

Reply:
done